### PR TITLE
Testing mailboxer

### PR DESCRIPTION
**Pull request by Sajaas**
* Specs for sending, receiving, replying & deleting message

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  describe 'DB table' do
+    it { is_expected.to have_db_column :name }
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :email }
+  end
+
+  describe 'mailboxer' do
+    let (:sender) { User.create(email: 'sender@test.com', password: 'password', name: 'sender') }
+    let (:receiver) { User.create(email: 'receiver@test.com', password: 'password', name: 'receiver') }
```
Comment (tochman) [inline spec/models/user_spec.rb]: This is good, but look into FactoryGirl. ;-)

File: spec/factories/user.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  factory :user do
+    sequence(:name) { |n| "user#{n}"}
+    email {"#{name}@test.com"}
+    password 'password'
+    end
+end
+
+
+# let(:sender) { User.create(email: 'sender@test.com', password: 'password', name: 'sender') }
+# let(:receiver) { User.create(email: 'receiver@test.com', password: 'password', name: 'receiver') }
```
Comment (tochman) [inline spec/factories/user.rb]: Why are these lines here?

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    let(:sender) {build(:user, name: 'sender') }
+    let(:receiver) {build(:user, name: 'receiver')}
+
+    before do
+      sender.send_message(receiver, "body", "subject")
+    end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+  end
```
Comment (tochman) [inline spec/models/user_spec.rb]: Move this outside this describe block. Not dealing with 'mailboxer'

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    end
+
+    describe 'send message' do
+
+      it 'should be a new message in the Recievers inbox' do
+        expect(receiver.mailbox.inbox.count).to eq 1
+      end
+
+      it 'should be a new message in the Senders outbox' do
+
```
Comment (tochman) [inline spec/models/user_spec.rb]: remove unnecessary empty lines

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      it 'should have a subject' do
+        expect(receiver.mailbox.inbox.last.subject).to eq "subject"
+      end
+
+      it 'should have a text body' do
+        conversation = receiver.mailbox.inbox.first
+        receipts = conversation.receipts_for receiver
+        receipts.each do |receipt|
+          @message = receipt.message
+        end
```
Comment (tochman) [inline spec/models/user_spec.rb]: Is this really needed? Why do you iterate over this instead of saying `@message = receipts.first.message`? Also, why do you need to store it in an instance variable?

**Review by tochman — CHANGES_REQUESTED** [review]
Getting there... Good job!

File: spec/factories/user.rb (hunk 2 above)
Comment (Sajaas) [reply spec/factories/user.rb]: Ooops I saved the first way just for reference. Forgot to remove it.

File: spec/models/user_spec.rb (hunk 5 above)
Comment (Sajaas) [reply spec/models/user_spec.rb]: I don't really know. This was the way Raoul showed us yesterday, quite frankly I don't understand it fully.

**Review by tochman — APPROVED** [review]
